contracts/src/system/contracts/TransactionPostProcessor.sol

```diff
@@ -32,18 +33,37 @@ contract TransactionPostProcessor is Initializable, AccessControl{
         _;
     }
 
-    IOnBlockEndCallback[] onBlockEndListeners;
+    IOnBlockEndCallback[] public onBlockEndListeners;
 
     function initialize(address eoaAdmin) public initializer {
         _grantRole(DEFAULT_ADMIN_ROLE, eoaAdmin);
         _grantRole(EOA_ADMIN_ROLE, eoaAdmin);
     }
 
-    function addOnBlockEndCallback(address callbackAddress) public onlyRole(EOA_ADMIN_ROLE) {
+    function addOnBlockEndCallback(address callbackAddress) external onlyRole(EOA_ADMIN_ROLE) {
+        require(callbackAddress != address(0), "Invalid callback address");
+        require(callbackAddress.code.length > 0, "Callback address must be a contract");
         onBlockEndListeners.push(IOnBlockEndCallback(callbackAddress));
+        emit CallbackAdded(callbackAddress);
+    }
+
+    function removeOnBlockEndCallback(address callbackAddress) external onlyRole(EOA_ADMIN_ROLE) {
```

Comment:
who is calling this?

Reply:
The EOA admin that we have registered when creating the network. So we can in practice stop any callback at anytime